src/configure/configure.ts

```diff
@@ -450,14 +450,34 @@ class PipelineConfigurer {
 
         // show available resources and get the chosen one
         this.appServiceClient = new AppServiceClient(this.inputs.azureSession.credentials, this.inputs.azureSession.tenantId, this.inputs.azureSession.environment.portalUrl, this.inputs.targetResource.subscriptionId);
+        
+        let resourceArray: Promise<Array<{label: string, data: GenericResource}>> = null;
+        let selectAppText: string = "";
+        let placeHolderText: string = "";
+
+        switch(this.inputs.pipelineParameters.pipelineTemplate.targetType) {
+            case TargetResourceType.WebApp:
+                resourceArray = this.appServiceClient.GetAppServices(this.inputs.pipelineParameters.pipelineTemplate.targetKind)
+                    .then((webApps) => webApps.map(x => { return { label: x.name, data: x }; }));
+                selectAppText = constants.SelectFunctionApp;
+                placeHolderText = Messages.selectFunctionApp;
+                break;
+            default:
+                resourceArray = this.appServiceClient.GetAppServices(WebAppKind.WindowsApp)
+                    .then((webApps) => webApps.map(x => { return { label: x.name, data: x }; }));
+                selectAppText = constants.SelectWebApp;
+                placeHolderText = Messages.selectWebApp;
+                break;
+        }
+
         let selectedResource: QuickPickItemWithData = await this.controlProvider.showQuickPick(
-            constants.SelectWebApp,
-            this.appServiceClient.GetAppServices(WebAppKind.WindowsApp)
-                .then((webApps) => webApps.map(x => { return { label: x.name, data: x }; })),
-            { placeHolder: Messages.selectWebApp },
+            selectAppText,
+            resourceArray,
+            { placeHolder:  placeHolderText },
             TelemetryKeys.WebAppListCount);
 
         this.inputs.targetResource.resource = selectedResource.data;
+        this.inputs.pipelineParameters.environment = selectedResource.label;
```

Comment:
this is same as resource.name. We do not need this as of now.

Reply:
This is needed in YAML Templates.